src/Microsoft.Extensions.Diagnostics.HealthChecks/DefaultHealthCheckService.cs

```diff
@@ -73,30 +73,39 @@ internal class DefaultHealthCheckService : HealthCheckService
                         try
                         {
                             var result = await healthCheck.CheckHealthAsync(context, cancellationToken);
+                            var duration = stopwatch.GetElapsedTime();
 
                             entry = new HealthReportEntry(
-                                result.Result ? HealthStatus.Healthy : registration.FailureStatus,
-                                result.Description,
-                                result.Exception,
-                                result.Data);
+                                status: result.Result ? HealthStatus.Healthy : registration.FailureStatus,
+                                description: result.Description,
+                                duration: duration,
+                                exception: result.Exception,
+                                data: result.Data);
 
-                            Log.HealthCheckEnd(_logger, registration, entry, stopwatch.GetElapsedTime());
+                            Log.HealthCheckEnd(_logger, registration, entry, duration);
                             Log.HealthCheckData(_logger, registration, entry);
                         }
 
                         // Allow cancellation to propagate.
                         catch (Exception ex) when (ex as OperationCanceledException == null)
                         {
-                            entry = new HealthReportEntry(HealthStatus.Failed, ex.Message, ex, data: null);
+                            entry = new HealthReportEntry(
+                                status: HealthStatus.Failed,
+                                description: ex.Message,
+                                duration: stopwatch.GetElapsedTime(),
+                                exception: ex,
+                                data: null);
+
```

Comment:
Sorry, forgot to comment here as well, this should also use a local for the time (L95 & L99)